<a href="https://colab.research.google.com/github/Florani1/First-Model/blob/main/Hamoye_Quiz_stage_A.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Food Balance Sheet Analysis**

In [ ]:
# import all packages and set plots to be displayed inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# ignore warning messages
import warnings
warnings.simplefilter("ignore")

In [ ]:
# import dataset to pandas dataframe

df = pd.read_csv("/content/FoodBalanceSheets_E_Africa_NOFLAG (1).csv", encoding = "latin-1")



In [ ]:
# check the total number of rows and columns in the dataframe
print(df.shape)

(60943, 12)


In [ ]:
# to see the first five rows in the dataset
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [ ]:
# to see the last five rows in the dataset
df.tail()

Area Code      Area  Item Code           Item  Element Code  \
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit  Y2014  Y2015  \
60938                                    Food      1000 tonnes  42.00  46.00   
60939     Food supply quantity (kg/capita/yr)               kg   3.06   3.33   
60940           Food supply (kcal/capita/day)  kcal/capita/day   3.00   4.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day   0.10   0.11   
60942      Fat supply quantity (g/capita/day)     g/capita/day   0.04   0.05   

       Y2016  Y2017  Y2018  
60938  33.00  19.00  16.00  
60939   2.35   1.33   1.08  
60940   3.00   1.00   1.00  
60941   0.08   0.04   0.04  
60942   0.03   0.02   0.01

In [ ]:
# to see all the attributes in the dataset
df.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

In [ ]:
# to get information about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [ ]:
# to see the distribution of the data
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [ ]:
df.describe(include='all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

In [ ]:
# to check for missing values
df.isna().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [ ]:
# to check for duplicate values
df.duplicated().any()

False

**QUESTIONS AND ANSWERS**

QUESTION 8
The total number of unique countries is 49.


QUESTION 12

The mean and standard deviation across the whole dataset for year 2017 is 140.917765 and 1671.862359	respectively.

Approximated to 2 d.p., we have :140.92 and 1671.86

In [ ]:
delf = df.groupby('Item')['Y2015','Y2018'].sum()
delf

<ipython-input-24-017dc29e1963>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  delf = df.groupby('Item')['Y2015','Y2018'].sum()


Y2015      Y2018
Item                                     
Alcohol, Non-Food      2180.00    2293.00
Alcoholic Beverages   98783.72   97847.27
Animal Products       11811.73   11578.61
Animal fats          200675.72  269648.27
Apples and products   10559.15    9640.51
...                        ...        ...
Vegetables, Other    158104.08  163987.21
Vegetal Products     107064.17  107775.39
Wheat and products   234710.51  242645.19
Wine                   4251.81    4039.32
Yams                 203151.78  221272.09

[119 rows x 2 columns]

Question 3 - 2017 has the highest sum of stock variation.

The codes are run below:

The maximum stock variation for year 2014 is 4569.

The maximum stock variation for year 2015 is 3418

The maximum stock variation for year 2016 is 3226

The maximum stock variation for year 2017 is 5089

The maximum stock variation for year 2018 is 2226

QUESTION 5:

Wine is	4251.81 for 2015 and	4039.32 for 2018

In [ ]:
elf.describe().Y2014

count          mean           std  \
Element                                                                      
Domestic supply quantity                5244.0    380.762080   2362.609680   
Export Quantity                         3660.0     40.989246    749.443225   
Fat supply quantity (g/capita/day)      4965.0      2.059529      7.796298   
Feed                                    1216.0    178.394646   1311.157481   
Food                                    4896.0    247.616930   1305.726276   
Food supply (kcal/capita/day)           4947.0     91.824742    356.558347   
Food supply quantity (kg/capita/yr)     4860.0     10.216179     29.871350   
Import Quantity                         4919.0     55.731750    483.260519   
Losses                                  2009.0     76.268293    371.204131   
Other uses (non-food)                   1662.0     47.363496    385.334275   
Processing                              1987.0    142.387016    910.888696   
Production                              3830.0    504.252676   2790.572788   
Protein supply quantity (g/capita/day)  4963.0      2.384941      8.928005   
Residuals                               4604.0      6.548436    349.266824   
Seed                                     762.0     28.770236     86.439248   
Stock Variation                         4230.0     13.888849    170.479352   
Total Population - Both sexes             45.0  22924.111111  31935.331040   
Tourist consumption                      555.0      0.749550      4.108235   

                                           min      25%       50%         75%  \
Element                                                                         
Domestic supply quantity                -133.0     0.00      6.00     79.0000   
Export Quantity                            0.0     0.00      0.00      3.0000   
Fat supply quantity (g/capita/day)         0.0     0.00      0.04      0.5100   
Feed                                       0.0     0.00      1.00     25.2500   
Food                                       0.0     0.00      5.00     60.0000   
Food supply (kcal/capita/day)              0.0     0.00      4.00     27.0000   
Food supply quantity (kg/capita/yr)        0.0     0.02      0.64      5.8125   
Import Quantity                            0.0     0.00      1.00      7.6000   
Losses                                     0.0     0.00      4.00     29.0000   
Other uses (non-food)                      0.0     0.00      0.00      1.8000   
Processing                                 0.0     0.00      0.00      5.0000   
Production                                 0.0     0.18     15.00    140.1350   
Protein supply quantity (g/capita/day)     0.0     0.00      0.03      0.6950   
Residuals                               -849.0     0.00      0.00      0.0000   
Seed                                       0.0     1.00      4.00     21.0000   
Stock Variation                        -1796.0     0.00      0.00      0.0000   
Total Population - Both sexes            196.0  3931.00  13664.00  26942.0000   
Tourist consumption                        0.0     0.00      0.00      0.0000   

                                              max  
Element                                            
Domestic supply quantity                108756.00  
Export Quantity                          39686.00  
Fat supply quantity (g/capita/day)          93.73  
Feed                                     29797.00  
Food                                     49014.00  
Food supply (kcal/capita/day)             3435.00  
Food supply quantity (kg/capita/yr)        428.27  
Import Quantity                          19582.00  
Losses                                    9807.00  
Other uses (non-food)                    12552.00  
Processing                               18383.00  
Production                              109593.00  
Protein supply quantity (g/capita/day)     100.34  
Residuals                                16697.00  
Seed                       

In [ ]:
elf.describe().Y2015

count          mean           std  \
Element                                                                      
Domestic supply quantity                5245.0    385.413451   2400.248993   
Export Quantity                         3704.0     42.552503    910.095249   
Fat supply quantity (g/capita/day)      4974.0      2.057849      7.801911   
Feed                                    1177.0    191.206644   1373.447354   
Food                                    4901.0    251.450949   1329.720375   
Food supply (kcal/capita/day)           4955.0     91.500101    356.076420   
Food supply quantity (kg/capita/yr)     4865.0     10.142884     29.625873   
Import Quantity                         4911.0     54.371505    467.826618   
Losses                                  2009.0     77.371329    376.049594   
Other uses (non-food)                   1667.0     39.744697    240.789221   
Processing                              1995.0    144.325313    920.450961   
Production                              3828.0    508.625755   2839.606113   
Protein supply quantity (g/capita/day)  4971.0      2.380597      8.932136   
Residuals                               4605.0      6.524430    337.553668   
Seed                                     759.0     31.590013    100.327105   
Stock Variation                         4229.0      8.255141    144.911156   
Total Population - Both sexes             45.0  23512.911111  32767.324536   
Tourist consumption                      555.0      0.628829      3.348996   

                                           min      25%       50%         75%  \
Element                                                                         
Domestic supply quantity                -122.0     0.00      6.00     79.0000   
Export Quantity                            0.0     0.00      0.00      2.9475   
Fat supply quantity (g/capita/day)         0.0     0.00      0.03      0.5000   
Feed                                       0.0     0.00      0.70     26.0000   
Food                                       0.0     0.00      5.00     61.0000   
Food supply (kcal/capita/day)              0.0     0.00      3.00     26.5000   
Food supply quantity (kg/capita/yr)        0.0     0.01      0.62      5.6800   
Import Quantity                            0.0     0.00      0.78      8.0000   
Losses                                     0.0     0.00      4.00     30.0000   
Other uses (non-food)                      0.0     0.00      0.00      2.0000   
Processing                                 0.0     0.00      0.00      5.0000   
Production                                 0.0     0.17     15.00    144.2500   
Protein supply quantity (g/capita/day)     0.0     0.00      0.03      0.6800   
Residuals                               -559.0     0.00      0.00      0.0000   
Seed                                       0.0     1.00      3.00     20.5000   
Stock Variation                        -3161.0     0.00      0.00      1.0000   
Total Population - Both sexes            199.0  4046.00  14111.00  27849.0000   
Tourist consumption                        0.0     0.00      0.00      0.0000   

                                              max  
Element                                            
Domestic supply quantity                110798.00  
Export Quantity                          52578.00  
Fat supply quantity (g/capita/day)          96.79  
Feed                                     30472.00  
Food                                     50705.00  
Food supply (kcal/capita/day)             3454.00  
Food supply quantity (kg/capita/yr)        418.21  
Import Quantity                          18824.00  
Losses                                    9999.00  
Other uses (non-food)                     6001.00  
Processing                               21609.00  
Production                              111663.00  
Protein supply quantity (g/capita/day)     100.33  
Residuals                                16136.00  
Seed                       

In [ ]:
elf.describe().Y2016

count          mean           std  \
Element                                                                      
Domestic supply quantity                5250.0    389.493848   2490.364824   
Export Quantity                         3653.0     41.587862    697.864440   
Fat supply quantity (g/capita/day)      4980.0      2.028669      7.686080   
Feed                                    1215.0    188.443333   1398.318948   
Food                                    4910.0    253.976002   1357.462687   
Food supply (kcal/capita/day)           4963.0     91.035664    355.039995   
Food supply quantity (kg/capita/yr)     4874.0     10.050324     29.451150   
Import Quantity                         4908.0     58.390949    508.556995   
Losses                                  2009.0     78.540070    387.422140   
Other uses (non-food)                   1634.0     42.572632    256.258796   
Processing                              1996.0    140.596693    907.334117   
Production                              3838.0    506.393213   2900.737754   
Protein supply quantity (g/capita/day)  4978.0      2.366350      8.900832   
Residuals                               4610.0      8.074620    379.056169   
Seed                                     759.0     30.815810     98.529837   
Stock Variation                         4231.0      7.832692    151.436282   
Total Population - Both sexes             45.0  24113.488889  33611.590685   
Tourist consumption                      555.0      0.160360      0.601934   

                                           min        25%       50%  \
Element                                                               
Domestic supply quantity                -167.0     0.0000      6.00   
Export Quantity                            0.0     0.0000      0.00   
Fat supply quantity (g/capita/day)         0.0     0.0000      0.03   
Feed                                       0.0     0.0000      0.00   
Food                                       0.0     0.0000      5.00   
Food supply (kcal/capita/day)              0.0     0.0000      3.00   
Food supply quantity (kg/capita/yr)        0.0     0.0100      0.59   
Import Quantity                            0.0     0.0000      1.00   
Losses                                     0.0     0.0000      4.00   
Other uses (non-food)                      0.0     0.0000      0.00   
Processing                                 0.0     0.0000      0.00   
Production                                 0.0     0.2525     15.00   
Protein supply quantity (g/capita/day)     0.0     0.0000      0.03   
Residuals                               -560.0     0.0000      0.00   
Seed                                       0.0     0.0000      3.00   
Stock Variation                        -3225.0     0.0000      0.00   
Total Population - Both sexes            203.0  4164.0000  14562.00   
Tourist consumption                        0.0     0.0000      0.00   

                                               75%        max  
Element                                                        
Domestic supply quantity                   79.0000  116483.00  
Export Quantity                             3.0000   35420.00  
Fat supply quantity (g/capita/day)          0.5000      94.33  
Feed                                       24.0000   31607.00  
Food                                       61.0000   52612.00  
Food supply (kcal/capita/day)              25.0000    3451.00  
Food supply quantity (kg/capita/yr)         5.6975     419.48  
Import Quantity                             8.0000   19858.00  
Losses                                     30.0000   10432.00  
Other uses (non-food)                       2.0000    6099.00  
Processing                                  5.0000   21739.00  
Production                                147.0000  119330.00  
Protein supply quantity (g/capita/day)      0.6575     100.94  
Residuals                                   0.0000   18077.00  
Seed                               

In [ ]:
elf.describe().Y2017

count          mean           std  \
Element                                                                      
Domestic supply quantity                5258.0    397.146843   2458.711288   
Export Quantity                         3660.0     49.819344    840.175529   
Fat supply quantity (g/capita/day)      4982.0      2.058177      7.817204   
Feed                                    1213.0    184.423479   1315.564296   
Food                                    4912.0    256.288331   1349.587482   
Food supply (kcal/capita/day)           4965.0     91.577241    357.340628   
Food supply quantity (kg/capita/yr)     4876.0      9.985652     29.260778   
Import Quantity                         4929.0     59.760416    482.217302   
Losses                                  2009.0     79.947237    393.408528   
Other uses (non-food)                   1619.0     56.606529    501.900270   
Processing                              1998.0    146.564565    966.799450   
Production                              3829.0    530.179391   3005.830342   
Protein supply quantity (g/capita/day)  4980.0      2.378002      8.942167   
Residuals                               4618.0      7.687311    366.797995   
Seed                                     758.0     32.810211    104.959459   
Stock Variation                         4231.0     12.837842    177.842748   
Total Population - Both sexes             45.0  24725.355556  34468.527183   
Tourist consumption                      555.0      0.163964      0.624528   

                                           min      25%        50%  \
Element                                                              
Domestic supply quantity                -144.0     0.00      6.235   
Export Quantity                            0.0     0.00      0.000   
Fat supply quantity (g/capita/day)         0.0     0.00      0.040   
Feed                                       0.0     0.00      0.200   
Food                                       0.0     0.00      5.000   
Food supply (kcal/capita/day)              0.0     0.00      3.000   
Food supply quantity (kg/capita/yr)        0.0     0.01      0.595   
Import Quantity                            0.0     0.00      1.000   
Losses                                     0.0     0.00      4.000   
Other uses (non-food)                      0.0     0.00      0.000   
Processing                                 0.0     0.00      0.000   
Production                                 0.0     0.49     15.720   
Protein supply quantity (g/capita/day)     0.0     0.00      0.030   
Residuals                               -682.0     0.00      0.000   
Seed                                       0.0     1.00      4.000   
Stock Variation                        -1582.0     0.00      0.000   
Total Population - Both sexes            207.0  4283.00  15017.000   
Tourist consumption                        0.0     0.00      0.000   

                                               75%        max  
Element                                                        
Domestic supply quantity                   81.0000  112625.00  
Export Quantity                             3.1350   34641.00  
Fat supply quantity (g/capita/day)          0.5075      97.13  
Feed                                       26.0000   29580.00  
Food                                       63.0000   51220.00  
Food supply (kcal/capita/day)              26.0000    3457.00  
Food supply quantity (kg/capita/yr)         5.7625     433.38  
Import Quantity                             8.0000   18627.00  
Losses                                     31.0000   10342.00  
Other uses (non-food)                       2.0000   17219.00  
Processing                                  5.0000   22174.00  
Production                                152.0000  117702.00  
Protein supply quantity (g/capita/day)      0.6400     100.10  
Residuals                                   0.0000   17476.00  
Seed                                       23.0000    1

The maximum stock variation for year 2014 is 4569.




The maximum stock variation for year 2015 is 3418



The maximum stock variation for year 2016 is 3226



The maximum stock variation for year 2017 is  5089



The maximum stock variation for year 2018 is 2226

QUESTION 11

Total number of the sum of processing is 292836.00

In [ ]:
df.groupby('Element')['Y2017'].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

In [ ]:
highest_sum = df.groupby('Area')['Y2017'].sum()

print(highest_sum)


Area
Algeria                         325644.27
Angola                          229159.57
Benin                           124771.22
Botswana                         22101.30
Burkina Faso                    101855.07
Cabo Verde                       14650.74
Cameroon                        232030.43
Central African Republic         29937.00
Chad                             71594.68
Comoros                             59.84
Congo                            41181.68
Côte d'Ivoire                   224599.01
Djibouti                         22729.91
Egypt                           866379.92
Eswatini                         54343.33
Ethiopia                        448683.76
Ethiopia PDR                         0.00
Gabon                            27979.64
Gambia                           23154.18
Ghana                           337599.06
Guinea                           98138.87
Guinea-Bissau                    19102.77
Kenya                           264660.66
Lesotho                      

In [ ]:
highest_sum('Area').max()

TypeError: ignored

In [ ]:
df.groupby('Element')['Element'].count()

Element
Domestic supply quantity                  5295
Export Quantity                           4403
Fat supply quantity (g/capita/day)        5023
Feed                                      1319
Food                                      4941
Food supply (kcal/capita/day)             5014
Food supply quantity (kg/capita/yr)       4905
Import Quantity                           5139
Losses                                    2009
Other uses (non-food)                     1732
Processing                                2010
Production                                3881
Protein supply quantity (g/capita/day)    5023
Residuals                                 4655
Seed                                       762
Stock Variation                           4232
Total Population - Both sexes               45
Tourist consumption                        555
Name: Element, dtype: int64

In [ ]:
print(df['Y2014'].to_string(index=False))

 38924.00
     0.00
  3377.00
    94.90
    80.06
  2932.00
    67.14
    53.84
   444.00
    27.76
    26.22
  3436.00
 12604.00
   684.00
     3.00
 15353.00
  4725.00
   177.00
  1084.00
    35.00
   787.00
     0.00
  8546.00
   219.56
  1680.00
    49.43
     6.84
  2436.00
  7461.00
   487.00
     3.00
  9407.00
   550.00
   123.00
   690.00
     0.00
   787.00
     0.00
  7258.00
   186.46
  1432.00
    43.20
     5.39
     0.00
   171.00
     8.00
      NaN
   163.00
     5.00
     0.00
     0.00
     0.00
     0.00
   158.00
     4.07
    27.00
     0.53
     0.04
   939.00
   805.00
    42.00
      NaN
  1702.00
  1028.00
    48.00
   103.00
    35.00
     0.00
   488.00
    12.54
    90.00
     2.46
     0.28
     3.00
  4148.00
   146.00
     0.00
  4005.00
  3077.00
     0.00
   288.00
     0.00
     0.00
   640.00
    16.44
   131.00
     3.23
     1.13
      NaN
     0.00
     0.00
     0.00
     0.00
     0.00
     0.00
     0.00
     0.00
    57.00
    13.00
     0.00
